### PR TITLE
core/gnmdriver: Proper interrupt registering

### PR DESCRIPTION
**Pull request by polybiusproxy**
Improves Dreams and Unreal Engine games.

### CODE REVIEW

**Comment by georgemoralis** [comment]
by improving we mean?

**Comment by polybiusproxy** [comment]
Games calling `sceGnmAddEqEvent` with another event ID (e.g. `Compute6RelMem` on Unreal Engine), then GfxEop or vice versa would assert and crash due to register of GfxEop event twice

File: src/core/platform.h [hunk 1]
```diff
@@ -81,7 +87,7 @@ struct IrqController {
         std::queue<IrqHandler> one_time_subscribers{};
         std::mutex m_lock{};
     };
-    std::array<IrqContext, magic_enum::enum_count<InterruptId>()> irq_contexts{};
+    std::unordered_map<InterruptId, IrqContext> irq_contexts{};
```
Comment (raphaelthegreat) [inline src/core/platform.h:90]: The number of interrupts is still quite small so an array will be faster than unoredered_map

Comment (polybiusproxy) [reply src/core/platform.h:90]: I don't see how to do this with an array here. What do you think would be the best approach?
I'm not sure of how much are these called so I don't know how vital performance is here

Comment (raphaelthegreat) [reply src/core/platform.h:90]: Before this change the array was indexed with InterruptId, now it was changed to hash map, so unless there is a new behavior that relies on the hashmap I don't see why abandoning array is needed

Comment (polybiusproxy) [reply src/core/platform.h:90]: I made it a map because I wasn't sure how to do indexing with the array without having a special if case for GfxEop

Comment (raphaelthegreat) [reply src/core/platform.h:90]: Can InterruptIdMax be increased to 0x41 so it works

**Pull request merged**